### DISCUSSION TITLE
How would you compile links into pdf?

### DISCUSSION THREAD

**Discussion by jorgemrisco** [artempyanykh/marksman#198] [Q&A] [answered]
We can link `md` to `md` using the `[[]]` syntax ("wiki-links").
It'd be nice to export markdowns to pdfs using `pandoc`. 
But wiki-links arent recognized and rendered as text only

Any ideas on how to approach this scenario?

**Comment by artempyanykh** [comment, marked as answer]
Hi @jorgemrisco! I was thinking about adding a command to compile the docs into vanilla markdown by rewriting wiki-links into regular markdown links, e.g.
```
marksman compile --out <folder>
```

Then you'd be able to feed the result of `marksman compile` into `pandoc`. How does this sound?

**Reply by jorgemrisco** [reply]
@artempyanykh sounds like a perfect solution to ensure compatibility!

**Reply by jorgemrisco** [reply]
@artempyanykh another interesting point in this discussion would be:
But then the compiled pdf would have links pointing to other markdown files and not other pdf.

Would you consider this out of scope?

**Comment by David-Else** [comment]
While it may be trickier to convert the wikilinks, you can filter them out using lua filters for pandoc. Here is how I filter hashtags:

```lua
-- hashtag-filter.lua
function Str(elem)
  -- Check if the string starts with a hashtag
  if elem.text:match("^#%w+") then
    -- Replace the hashtag with an empty string
    return pandoc.Str(elem.text:gsub("^#%w+", ""))
  end
end
```

To use this filter, save the Lua code to a file named `hashtag-filter.lua` (in your pandoc filters dir) and then run Pandoc with the `--lua-filter` option:

```bash
pandoc input.md --lua-filter=hashtag-filter.lua -o output.html
```

This command will convert `input.md` to `output.html` and apply the Lua filter to remove hashtags.
